tmt/trying.py

```diff
@@ -0,0 +1,294 @@
+""" Easily try tests and experiment with guests """
+
+import textwrap
+from typing import Any, List
+
+import click
+import fmf
+from fmf.utils import listed
+
+import tmt
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt import Plan
+from tmt.templates import DEFAULT_PLAN, DEFAULT_PLAN_NAME
+from tmt.utils import MetadataError, Path
+
+USER_PLAN_NAME = "/user/plan"
+
+
+class Try(tmt.utils.Common):
+
+    def __init__(
+            self,
+            *,
+            tree: tmt.Tree,
+            logger: tmt.log.Logger,
+            **kwargs: Any) -> None:
+        """ Just store the tree """
+
+        super().__init__(logger=logger, **kwargs)
+
+        self.tree = tree
+        self.tests: List[tmt.Test] = []
+        self.plans: List[Plan] = []
+        self.image_and_how = self.opt("image_and_how")
+
+    def check_tree(self) -> None:
+        """ Make sure there is a sane metadata tree """
+
+        # Both tree and root should be defined
+        try:
+            if self.tree and self.tree.root:
+                return
+
+        # Create a dumb fmf Tree if no metadata around
+        except MetadataError:
+            new_fmf_tree = fmf.Tree({"nothing": "here"})
+            self.tree.tree = new_fmf_tree
+
+    def check_tests(self) -> None:
+        """ Check for available tests """
+
+        # Search for tests according to provided names
+        test_names = list(self.opt("test"))
+        if test_names:
+            self.tests = self.tree.tests(names=test_names)
+            if not self.tests:
+                raise tmt.utils.GeneralError(
+                    f"No test matching '{listed(test_names)}' found.")
+
+        # Default to tests under the current working directory
+        else:
+            if not self.tree.root:
+                self.debug("No fmf tree root, no tests.")
+                return
+            relative_path = Path(".").relative_to(self.tree.root)
+            test_names = [f"^/{relative_path}"]
+            self.tests = self.tree.tests(names=test_names)
```

Comment:
Good point. I think we do not have to bail out completely here but just give warning to the user. It might be a perfectly valid use case to enter a an empty directory and play a bit with something using `login`. Should be fixed by 374cad95.